storage/remote/client_test.go

```diff
@@ -0,0 +1,71 @@
+package remote
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+)
+
+func TestStoreHTTPErrorHandling(t *testing.T) {
+	tests := []struct {
+		code        int
+		shouldFail  bool
+		recoverable bool
```

Comment:
You could pass in the expected error as part of the test case, then the checks would be a single `reflect.DeepEqual`.